src/GerdaFitter.cc

```diff
@@ -1124,3 +1189,82 @@ TF1 GerdaFitter::ParseTFormula(std::string prefix, std::string expr, double rang
         return _tformula;
     }
 }
+
+template<typename BasicJsonType>
+std::vector<std::pair<double,double>> GerdaFitter::CheckAndStoreRanges(BasicJsonType& range) {
+    bool _bad_range = false;
+    std::vector<std::pair<double,double>> _v_range;
+    if (range.is_array()) {
+        if (range[0].is_array()) {
+            for (auto _r : range) {
+                if (_r[0].is_number() and _r[1].is_number()) {
+                    _v_range.push_back({_r[0],_r[1]});
+                }
+                else _bad_range = true;
+            }
+        }
+        else if (range[0].is_number() and range[1].is_number()) {
+            _v_range.push_back({range[0],range[1]});
+        }
+        else _bad_range = true;
+    }
+    else _bad_range = true;
+
+    if (_bad_range) throw std::runtime_error("Range is ill-defined");
+
+    return _v_range;
+}
+
+std::vector<std::pair<int,int>> GerdaFitter::TranslateAxisRangeToBinRange(
+    TH1* /*h*/,
+    std::vector<std::pair<double,double>> /*x_range*/,
+    std::vector<std::pair<double,double>> /*y_range*/
+) {
+    BCLog::OutSummary("Implement me : TranslateAxisRangeToBinRange");
+    std::vector<std::pair<int,int>> _b_range;
+    return _b_range;
+}
+
+double GerdaFitter::IntegrateHistogram(
+    TH1* h,
+    std::vector<std::pair<double,double>> x_range,
+    std::vector<std::pair<double,double>> y_range
+) {
+    if (h->GetDimension() > 2) {
+        throw std::runtime_error("IntegrateHistogram not implemeted for TH3.");
+    }
+    else if (h->GetDimension() == 2) {
+        return IntegrateHistogram2D(dynamic_cast<TH2*>(h), x_range, y_range);
+    }
+    else if (h->GetDimension() == 1) {
+        return IntegrateHistogram1D(h, x_range);
+    }
+    else {
+        throw std::runtime_error("Something went wrong in IntegrateHistogram.");
+    }
+    return 0.;
+}
+
+double GerdaFitter::IntegrateHistogram1D(TH1* h, std::vector<std::pair<double,double>> range) {
+    double integral = 0.;
+    for (auto _r : range) {
+        int _b_min = h->FindBin(_r.first);
+        int _b_max = h->FindBin(_r.second);
+        BCLog::OutDebug(" -> IntegrateHistogram1D ["
+            + std::to_string(_b_min) + "," + std::to_string(_b_max)
+            + "] n-bins : " + std::to_string(_b_max-_b_min+1)
+        );
+        integral += h->Integral(_b_min, _b_max);
+    } 
+    return integral;
+}
+
+double GerdaFitter::IntegrateHistogram2D(
+    TH2* /*h*/,
+    std::vector<std::pair<double,double>> /*x_range*/,
+    std::vector<std::pair<double,double>> /*y_range*/
+)
+{ 
+    BCLog::OutSummary("Implement me : IntegrateHistogram::TH2");
+    return 0;
+}
```

Comment:
Queste non possono essere eliminate?

Reply:
IntegrateHistogram1D viene usato quindi no. IntegrateHistogram2D ho messo se in futuro vuoi usarlo.